jwala-services/src/main/java/com/cerner/jwala/service/group/impl/GroupServiceImpl.java

```diff
@@ -411,7 +416,13 @@ protected CommandOutput executeDeployGroupAppTemplate(final String groupName, fi
                 .setGroupName(groupName)
                 .setWebAppName(application.getName())
                 .build();
-        return resourceService.generateAndDeployFile(resourceIdentifier, application.getName(), fileName, hostName);
+        String lockKey = groupName + application.getName() + fileName + JwalaUtils.getHostAddress(hostName);
```

Comment:
This already done, where ever there was a host used in the key, it has been changed to IP address. Look at ApplicationServiceImpl.java